provider/database_db_homes_resource_test.go

```diff
@@ -0,0 +1,216 @@
+	}
+	db_system_id = "${var.db_system_id}"
+	db_version = "${var.db_version}"
+}
+
+	`
+
+	s.Config += testProviderConfig()
+
+	s.ResourceName = "oci_database_db_home.t"
+
+}
+
+func (s *ResourceDBHomeTestSuite) TestCreateResourceDBHome() {
+
+	resource.UnitTest(s.T(), resource.TestCase{
+		Providers: s.Providers,
+		Steps: []resource.TestStep{
+			{
+				ImportState:       true,
+				ImportStateVerify: true,
+				Config:            s.Config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(s.ResourceName, "display_name", "display_name"),
+					resource.TestCheckResourceAttrSet(s.ResourceName, "id"),
+				),
+			},
+		},
+	})
+}
+
+func (s ResourceDBHomeTestSuite) TestUpdateDBHomeDBVersion() {
+	config := `
+resource "oci_database_db_home" "testDBHome" {
+	#Required
+	database {
+		#Required
+		admin_password = "${var.database_admin_password}"
+		db_name = "${var.database_db_name}"
+	}
+	db_system_id = "${var.db_system_id}"
+	db_version = "${var.db_version}"
+}
+`
+
+	config += testProviderConfig()
+
+	resource.UnitTest(s.T(), resource.TestCase{
+		Providers: s.Providers,
+		Steps: []resource.TestStep{
+			{
+				ImportState:       true,
+				ImportStateVerify: true,
+				Config:            s.Config,
+			},
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(s.ResourceName, "db_version", "newValue"),
+				),
+			},
+		},
+	})
+}
+
+func (s ResourceDBHomeTestSuite) TestUpdateDatabaseForcesNewDBHome() {
+
+	config := `
+resource "oci_database_db_home" "testDBHome" {
+	#Required
+	database {
+		#Required
+		admin_password = "${var.database_admin_password}"
+		db_name = "${var.database_db_name}"
+	}
+	db_system_id = "${var.db_system_id}"
+	db_version = "${var.db_version}"
+}
+`
+
+	config += testProviderConfig()
+
+	resource.UnitTest(s.T(), resource.TestCase{
+		Providers: s.Providers,
+		Steps: []resource.TestStep{
+			{
+				ImportState:       true,
+				ImportStateVerify: true,
+				Config:            s.Config,
+			},
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(s.ResourceName, "database", "newValue"),
+				),
+			},
+		},
+	})
+}
+
+func (s ResourceDBHomeTestSuite) TestUpdateDBSystemIDForcesNewDBHome() {
+
+	config := `
+resource "oci_database_db_home" "testDBHome" {
+	#Required
+	database {
+		#Required
+		admin_password = "${var.database_admin_password}"
+		db_name = "${var.database_db_name}"
+	}
+	db_system_id = "${var.db_system_id}"
+	db_version = "${var.db_version}"
+}
+`
+
+	config += testProviderConfig()
+
+	resource.UnitTest(s.T(), resource.TestCase{
+		Providers: s.Providers,
+		Steps: []resource.TestStep{
+			{
+				ImportState:       true,
+				ImportStateVerify: true,
+				Config:            s.Config,
+			},
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(s.ResourceName, "db_system_id", "newValue"),
```

Comment:
Does this test work? Where is newValue set?

Reply:
newValue is meant to be replaced manually by an actual valid value